pkg/ddl/ingest/backend_mgr.go

```diff
@@ -109,26 +132,57 @@ func (b *BackendCtxBuilder) Build(cfg *local.BackendConfig, bd *local.Backend) (
 
 	//nolint: forcetypeassert
 	pdCli := store.(tikv.Storage).GetRegionCache().PDClient()
-	var cpMgr *CheckpointManager
-	if b.sessPool != nil {
-		cpMgr, err = NewCheckpointManager(ctx, b.sessPool, b.physicalID, job.ID, jobSortPath, pdCli)
-		if err != nil {
-			logutil.Logger(ctx).Warn("create checkpoint manager failed",
-				zap.Int64("jobID", job.ID),
-				zap.Error(err))
-			return nil, err
+	var cpOp CheckpointOperator
+
+	// Create checkpoint manager based on the configuration
+	if b.useDistTask {
+		// Use distributed task checkpoint manager
+		if b.updateFunc != nil || b.getFunc != nil {
```

Comment:
Can we assume if `b.useDistTask != nil`, both `updateFunc` and `getFunc` are not nil?